src/libproc_macro/lib.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! ## Example: Implementing Scheme's `cond`
```

Comment:
Could you add more informational documentation here please? For example, the model of tokens represented by TokenStreams, using quote to make new tokens. I don't think you need such a  big example - maybe just reference the test file with it in.